src/emulator.c

```diff
@@ -1326,6 +1520,12 @@ void ofono_emulator_set_indicator(struct ofono_emulator *em,
 	gboolean call;
 	gboolean callsetup;
 	gboolean waiting;
+	struct ofono_emulator *em = __ofono_atom_get_data(atom);
+
+	if (!valid_indication(em, atom, name))
```

Comment:
I would have simply added the "struct ofono_atom" as a function parameter instead of replacing "struct ofono_emulator", but seemed to me senseless as you can obtain the emulator from the atom.

Changing the name to ofono_atom_*() does not really make sense either.

To really fix this we would need a real implementation of the hfp atom (src/hfp.c?) instead of it being a dummy atom that points to the emulator. Each modem would have an instance of "struct ofono_hfp", and the instances would point to a the single emulator instance. But that seemed to me a too big change.